R/SEIRmodels.R

```diff
@@ -0,0 +1,749 @@
+      legend.box = c("horizontal", "vertical")
+    ) +
+    ggplot2::scale_colour_manual(
+      values = c("blue", "red", "green", "purple", "yellow")) +
+    ggplot2::theme(
+      text = ggplot2::element_text(color = "#444444", family = "Lucida Bright"),
+      plot.title = ggplot2::element_text(size = 26, color = "#333333"),
+      plot.subtitle = ggplot2::element_text(size = 13),
+      axis.title.x = ggplot2::element_text(size = 16, color = "#333333"),
+      axis.title.y = ggplot2::element_text(angle = 0, vjust = .5))
+}
+
+Displaythemodel3 <- function(df) {
+
+  # This function displays data in a list. df must be in the form of a list.
+
+  # Check if df is a dataframe. If yes then turn it into a list
+  numruns <- 0
+  datapoints <- 0
+  subtitle <- ""
+
+  if (is.data.frame(df)) {
+    numdatapoints <- paste(length(df$time))
+    numruns <- 1
+    df <- list(df)
+    subtitle <- paste(
+      "Simulation for", numruns, "run and", numdatapoints, "data points")
+  } else {
+    numruns <- length(df)
+    numdatapoints <- length(df[[1]][[1]]) - 1
+    subtitle <- paste(
+      "Simulation for", numruns, "runs,", numdatapoints, "data points per run")
+  }
+
+  # Create group id for data
+  df <- dplyr::bind_rows(df, .id = "group")
+
+  #Convert to long format
+  df <- tidyr::pivot_longer(tibble::as_tibble(df),
+                            c("S", "E1", "E2", "IMild", "ICase1", "ICase2",
+                              "cum_hosp_inc", "IOxGetLive1", "IOxGetLive2",
+                              "IOxNotGetLive1", "IOxNotGetLive2", "IOxGetDie1",
+                              "IOxGetDie2", "IOxNotGetDie1", "IOxNotGetDie2",
+                              "IMVGetLive1", "IMVGetLive2", "IMVNotGetLive1",
+                              "IMVNotGetLive2", "IMVGetDie1", "IMVGetDie2",
+                              "IMVNotGetDie1", "IMVNotGetDie2", "IRec1",
+                              "IRec2", "R", "D"))
+
+  #df <- tidyr::pivot_longer(tibble::as_tibble(df), cols = df$nameslist)
+
+  strname <- paste("SIR", df$type, "Model Simulation")
+
+  ggplot2::ggplot(
+                  df, ggplot2::aes(x = df$time, y = df$value,
+                  group = interaction(df$group, df$name), colour = df$name)) +
+    ggplot2::geom_line(size = 0.5) +
+    ggplot2::theme_bw() +
+    ggplot2::labs(title = strname, subtitle = subtitle, color = df$legend) +
+    ggplot2::labs(y = "States", x = "time") +
+    ggplot2::theme(
+      legend.justification = c("right", "top"),
+      legend.box = c("horizontal", "vertical")
+    ) +
+    ggplot2::theme(
+      text = ggplot2::element_text(color = "#444444", family = "Lucida Bright"),
+      plot.title = ggplot2::element_text(size = 26, color = "#333333"),
+      plot.subtitle = ggplot2::element_text(size = 13),
+      axis.title.x = ggplot2::element_text(size = 16, color = "#333333"),
+      axis.title.y = ggplot2::element_text(angle = 0, vjust = .5))
+
+}
+
+
+#' @title Infection_process -> S to I
```

Comment:
I was confused about what this function returned. Maybe this would help?

```suggestion
#' @title Generates an infection_process -> S to I
```